schematic/models/validate_manifest.py

```diff
@@ -174,7 +160,8 @@ def validate_manifest_rules(
         warnings = []
 
         if not restrict_rules:
-            t_GE = perf_counter()
+            if logger.isEnabledFor(logging.DEBUG):
+                t_GE = perf_counter()
```

Comment:
Some extra code to help guard against extra things from being tracked in memory (It's super minor)